stdlib/ballerina-builtin/src/main/ballerina/ballerina/collections/vector.bal

```diff
@@ -0,0 +1,147 @@
+
+    if (err != null) {
+        return null, err;
+    }
+
+    return v.vec[index], null;
+}
+
+@Description { value:"Inserts the given element at the position specified. All the elements (including the one currently in the position specified) to the right of the specified position are shifted to the right."}
+@Param { value: "v: The vector to which the element will be inserted"}
+@Param { value: "element: The element to insert"}
+@Param { value: "index: The position to insert the element to"}
+@Return { value:"Returned if the specified index is not within 0 and the size of the vector (both inclusive)."}
+public function <Vector v> insert (any element, int index) (IndexOutOfRangeError) {
+    IndexOutOfRangeError err = validateRange(v.size + 1, index); // range validated for the new vector size
+
+    if (err != null) {
+        return err;
+    }
+
+    shiftRight(v, index);
+    v.vec[index] = element;
+    v.size = v.size + 1;
+    return null;
+}
+
+@Description { value:"Removes and returns the element at the position specified. All the elements to the right of the specified position are shifted to the left."}
+@Param { value: "v: The vector from which the element will be removed"}
+@Param { value: "index: The position to remove the element from"}
+@Return { value:"The element at the specified position."}
+@Return { value:"Returned if the specified index is not within 0 (inclusive) and the size of the vector (exclusive)."}
+public function <Vector v> remove (int index) (any, IndexOutOfRangeError) {
+    IndexOutOfRangeError err = validateRange(v.size, index);
+
+    if (err != null) {
+        return null, err;
+    }
+
+    any element = v.vec[index];
+    shiftLeft(v, index);
+    v.size = v.size - 1;
+    return element, null;
+}
+
+@Description { value:"Replaces the element at the position specified with the provided element."}
+@Param { value: "v: The vector in which the element will be replaced"}
+@Param { value: "element: The replacement element"}
+@Param { value: "index: The position of the element to be replaced"}
+@Return { value:"The element which was originally at the specified position"}
+@Return { value:"Returned if the specified index is not within 0 (inclusive) and the size of the vector (exclusive)."}
+public function <Vector v> replace(any element, int index) (any, IndexOutOfRangeError) {
+    IndexOutOfRangeError err = validateRange(v.size, index);
+
+    if (err != null) {
+        return null, err;
+    }
+
+    any currentElement = v.vec[index];
+    v.vec[index] = element;
+
+    return currentElement, null;
+}
+
+@Description { value:"Returns the size of the vector."}
+@Param { value: "v: The vector of which to look-up the size"}
+@Return { value:"The size of the vector"}
+public function <Vector v> size() (int) {
+    return v.size;
+}
+
+function shiftRight (Vector v, int index) {
+    v.vec[v.size] = null; // assigning null so that v.size-th array index is valid
```

Comment:
This algorithm can be written in a simple manner. Check out the following code block.

```ballerina
v.vec[v.size] = null; // assigning null so that v.size-th array index is valid
int i = v.size;

while (index < i ) {
    v[i] = v[i -1];
    i -=1;
}

```

Reply:
Ack. Will change this.